src/ORM/FieldType/DBForeignKey.php

```diff
@@ -87,12 +87,27 @@ public function scaffoldFormField($title = null, $params = null)
 
         // Add the count of the list to a cache for subsequent calls
         if (!isset(static::$foreignListCache[$hasOneClass])) {
+            // Let the DB do the threshold check as it will be faster - depending on the SQL engine it might only have
+            // to count indexes
+            $dataQuery = $list->dataQuery()->getFinalisedQuery();
+
+            // Clear order-by as it's not relevant for counts
+            $dataQuery->setOrderBy(false);
+            // Remove distinct. Applying distinct shouldn't be required provided relations are not applied.
+            $dataQuery->setDistinct(false);
+
+            $dataQuery->setSelect(['over_threshold' => 'count(*) > ' . (int) $threshold]);
+            $result = $dataQuery->execute()->column('over_threshold');
+            $overThreshold = !empty($result) && ($result[0] === 't' ||  (int) $result[0] === 1);
+
             static::$foreignListCache[$hasOneClass] = [
-                'count' => $list->count(),
+                'overThreshold' => $overThreshold,
```

Comment:
It's possible albeit incredibly unlikely that this is a breaking change if someone is relying on the existing `count` cache in a subclass.

Reply:
I'm not sure we'd consider that part of our API to be honest